src/filesystem.cc

```diff
@@ -1473,22 +1507,46 @@ S3FileSystem::S3FileSystem(
     }
   }
 
-  if ((secret_key != NULL) && (key_id != NULL)) {
-    client_ = s3::S3Client(
+  if (secret_key.size() && key_id.size()) {
```

Comment:
`!empty()` instead of `size()`, apply everywhere applicable

Reply:
Updated.